tests/e2e/pageobjects/git-providers/OauthPage.ts

```diff
@@ -17,140 +17,186 @@ import { TimeoutConstants } from '../../constants/TimeoutConstants';
+          this.submitButton =
+            TestConstants.TS_SELENIUM_GIT_PROVIDER_IS_LDAP_LOGIN
+              ? By.xpath('//input[@data-qa-selector="sign_in_button"]')
+              : By.xpath('//button[@data-qa-selector="sign_in_button"]');
+          this.approveButton = By.xpath('//*[@value="Authorize"]');
+          this.denyAccessButton = By.xpath('//input[@value="Deny"]');
+          this.twoFactorInfoBox = By.xpath('input[@id="user_otp_attempt"]');
+        }
+        break;
+      case GitProviderType.GITHUB:
+        {
+          this.loginForm = By.id('login_field');
+          this.passwordForm = By.id('password');
+          this.approveButton = By.xpath('//*[@id="js-oauth-authorize-btn"]');
+          this.submitButton = By.xpath('//*[@value="Sign in"]');
+          this.denyAccessButton = By.xpath('//button[contains(., "Cancel")]');
+          this.twoFactorInfoBox = By.xpath(
+            '//div[contains(@class, "auth-form-header")]'
+          );
+        }
+        break;
+      default: {
+        throw new Error(
+          `Invalid git provider. The value should be ${GitProviderType.GITHUB}, ${GitProviderType.GITLAB} or ${GitProviderType.BITBUCKET}`
+        );
+      }
+    }
+  }
+
+  async waitLoginPage(): Promise<void> {
+    Logger.debug('OauthPage.waitLoginPage');
+
+    // for gitlab server https://gitlab.cee.redhat.com
+    if (
+      !TestConstants.TS_SELENIUM_GIT_PROVIDER_IS_LDAP_LOGIN &&
+      TestConstants.TS_SELENIUM_FACTORY_GIT_PROVIDER === GitProviderType.GITLAB
+    ) {
+      {
+        Logger.debug(
+          `OauthPage.login - go to ${TestConstants.TS_SELENIUM_FACTORY_GIT_PROVIDER} Standard login section`
+        );
+        await this.driverHelper.waitAndClick(
+          By.xpath('//a[@data-qa-selector="standard_tab"]')
+        );
+      }
     }
 
-    async waitLoginPage(): Promise<void> {
-        Logger.debug('OauthPage.waitLoginPage');
+    await this.driverHelper.waitVisibility(
+      this.loginForm,
+      TimeoutConstants.TS_SELENIUM_CLICK_ON_VISIBLE_ITEM * 3
+    );
+  }
 
-        // for gitlab server https://gitlab.cee.redhat.com
-        if (!TestConstants.TS_SELENIUM_GIT_PROVIDER_IS_LDAP_LOGIN &&
-            TestConstants.TS_SELENIUM_FACTORY_GIT_PROVIDER === GitProviderType.GITLAB) {
-            {
-                Logger.debug(`OauthPage.login - go to ${TestConstants.TS_SELENIUM_FACTORY_GIT_PROVIDER} Standard login section`);
-                await this.driverHelper.waitAndClick(By.xpath('//a[@data-qa-selector="standard_tab"]'));
-            }
-        }
+  async waitDisappearanceTwoFactorInfoBox(): Promise<void> {
+    Logger.debug(`OauthPage.waitDisappearanceTwoFactorInfoBox`);
 
-        await this.driverHelper.waitVisibility(this.loginForm, TimeoutConstants.TS_SELENIUM_CLICK_ON_VISIBLE_ITEM * 3);
-    }
+    await this.driverHelper.waitDisappearance(
+      this.twoFactorInfoBox,
+      TimeoutConstants.TS_SELENIUM_GIT_TWO_FACTOR_TIMEOUT,
```

Comment:
This variable describes number of attempts to use with polling. If you want to use it as a timeout, I'd rather consider calculating it locally in the `waitDisappearanceTwoFactorInfoBox` method as follows:
```
const attempts: number = TimeoutConstants.TS_SELENIUM_GIT_TWO_FACTOR_TIMEOUT / polling;
```
and then use the `attempts` local variable

Reply:
Thanks for the review, I've taken the polling rate into account and I've rewritten it.

Reply:
ACK, thank you.